bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -458,7 +458,7 @@ type KubeadmConfigStatus struct {
 
 	// dataSecretName is the name of the secret that stores the bootstrap data script.
 	// +optional
-	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MinLength=0
```

Comment:
Just to double check. Do you really need this here as well?

I thought you are setting dataSecretName: "" *instead* of referencing a bootstrap object